pkg/services/business_case.go

```diff
@@ -280,3 +280,37 @@ func NewUpdateBusinessCase(
 		return businessCase, nil
 	}
 }
+
+// NewArchiveBusinessCase is a service to archive a businessCase
+func NewArchiveBusinessCase(
+	config Config,
+	fetch func(id uuid.UUID) (*models.BusinessCase, error),
+	update func(*models.BusinessCase) (*models.BusinessCase, error),
+) func(context.Context, uuid.UUID) error {
+	return func(ctx context.Context, id uuid.UUID) error {
+		businessCase, fetchErr := fetch(id)
+		if fetchErr != nil {
+			return &apperrors.QueryError{
+				Err:       fetchErr,
+				Operation: apperrors.QueryFetch,
+				Model:     businessCase,
+			}
+		}
+
+		updatedTime := config.clock.Now()
```

Comment:
Is is valid to archive something that's already been archived? 

Also, this might be a question that leads into the state machine work, but are there other transitions that are invalid to/from Archived?

Reply:
all states should be able to move to archived, but as of right now there are no paths from archived as a status.